MovingBlocks/Terasology
# Replace visual character and allow for replacement by modules

### Contains

This PR:
* This PR introduces a new event (CreateVisualCharacterEvent) that modules can handle to create their own character visualization 
  * The visualization is completely up to the module. It could be a static model, a animated one or theoreticlly just a particle system.
  * The model can but does not need to be colored like the player. It is up to the event handler
  * For the own character no CreateVisualCharacterEvent gets fired so that "hideFromOwner" flag is needed for the SkeletalMeshComponent 
  * If the event does not get handled, a floating cube will be shown as character model (instead of the moneky head)
* The existing monkey head has been removed as it has been replaced by the floating cube.

The reason for replacing the default monkey head with a default floating cube is that the monkey head is not so clearly a place holder and people would just wonder why there is a monkey head. Also it is better to show of an animated model instead of a static one.

### How to test
Unit tests:
* Run VisualCharacterSystemTest

Manual testing:
1. Start the game as listen server.
2. Start another instance of the game and connect to server
3. Expected result: The own character is invisible in each instance while the other is a cube floating slightly and down constantly.
